zebra-chain/src/transaction/sighash.rs

```diff
@@ -48,6 +52,23 @@ impl HashType {
     }
 }
 
+/// A Signature Hash (or SIGHASH) as specified in
+/// https://zips.z.cash/protocol/protocol.pdf#sighash
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Serialize, Deserialize, Debug)]
+pub struct Hash(pub [u8; 32]);
```

Comment:
So, the proper name is literally [`SIGHASH`](https://zips.z.cash/protocol/protocol.pdf#sighash), and I don't think I ever hear anyone say 'signature hash' when they are talking about the result of applying the `SIGHASH` algorithm to a transaction, so we may just want to call it `SigHash` everywhere

Reply:
I was confused about the terminology because Bitcoin seems to call ["sighash" the sighash flags](https://developer.bitcoin.org/glossary.html#term-Sighash), not the hash itself (which is called "Signed hash"?). ZIP-244 also uses "signature digest".

But yeah, using yet another name is even more confusing :sweat_smile: , I've changed it to `SigHash`. Done in 463f5f29

Reply:
There's so much overloading of all these terms over the history of Bitcoin and now Zcash 🙃 📜 📚